qiskit/utils/deprecation.py

```diff
@@ -434,3 +443,29 @@ def add_deprecation_to_docstring(
     else:
         original_lines.extend(new_lines)
     func.__doc__ = "\n".join(original_lines)
+
+
+def remove_sphinx_from_runtime_deprecation_warning(msg: str) -> str:
+    """Remove Sphinx-isms from ``msg`` so that it's more readable in runtime warnings.
+
+    This function is useful so that you can write deprecation messages using Sphinx syntax like
+    ``:class:`` references, so that the deprecation renders more nicely in docs. But, that syntax
+    is not very readable when we show the same message as a runtime warning, so this function
+    allows you to strip it before calling ``warnings.warn()``.
+
+    This function will:
+
+    * Replace `` with `
+    * Remove roles like `:ref:`, `:class:`, and `:py:func:`.
```

Comment:
I'm surprised that these two lines don't break the build tbh.  You may also be missing doubled backticks on line 459.